src/diffpy/labpdfproc/tools.py

```diff
@@ -28,6 +29,41 @@ def set_output_directory(args):
     return output_dir
 
 
+def expand_wildcard_file(args):
+    """
+    Expands wildcard inputs by adding all files or directories within directories matching the pattern.
+
+    Parameters
+    ----------
+    args argparse.Namespace
+        the arguments from the parser
+
+    Returns
+    -------
+    the arguments with the wildcard inputs expanded
+
+    """
+    wildcard_inputs = [input_name for input_name in args.input if "*" in input_name]
+    for wildcard_input in wildcard_inputs:
+        if not glob.glob(wildcard_input):
+            raise FileNotFoundError(
+                f"Invalid wildcard input {wildcard_input}. "
+                f"Please ensure the wildcard pattern matches at least one file or directory."
+            )
+        input_files = Path(".").glob(wildcard_input)
+        for input_file in input_files:
+            if input_file.is_file():
```

Comment:
I think some of the logic here is handled later.  I would just operate on strings here.
```
for wildcard_input in wildcard_inputs:
    input_files = Path(".").glob(wildcard_input)
    args.input.extend(input_files)
    args.input.remove(wildcard_input)
```
and that's it?